seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcGreenplumIT.java

```diff
@@ -0,0 +1,151 @@
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+@Slf4j
+public class JdbcGreenplumIT extends FlinkContainer {
+
+    private static final String GREENPLUM_IMAGE = "datagrip/greenplum:6.8";
+    private static final String GREENPLUM_CONTAINER_HOST = "flink_e2e_greenplum";
+    private static final int GREENPLUM_CONTAINER_PORT = 5432;
+    private static final String GREENPLUM_HOST = "localhost";
+    private static final int GREENPLUM_PORT = 5435;
+    private static final String GREENPLUM_USER = "tester";
+    private static final String GREENPLUM_PASSWORD = "pivotal";
+    private static final String GREENPLUM_DRIVER = "org.postgresql.Driver";
+    private static final String GREENPLUM_JDBC_URL = String.format(
+            "jdbc:postgresql://%s:%s/testdb", GREENPLUM_HOST, GREENPLUM_PORT);
+
+    private GenericContainer<?> greenplumServer;
+    private Connection jdbcConnection;
+
+    @BeforeEach
+    public void startGreenplumContainer() throws ClassNotFoundException, SQLException {
+        greenplumServer = new GenericContainer<>(GREENPLUM_IMAGE)
+                .withNetwork(NETWORK)
+                .withNetworkAliases(GREENPLUM_CONTAINER_HOST)
+                .withLogConsumer(new Slf4jLogConsumer(log));
+        greenplumServer.setPortBindings(Lists.newArrayList(
+                String.format("%s:%s", GREENPLUM_PORT, GREENPLUM_CONTAINER_PORT)));
+        Startables.deepStart(Stream.of(greenplumServer)).join();
+        log.info("Greenplum container started");
+        // wait for Greenplum fully start
+        Class.forName(GREENPLUM_DRIVER);
+        given().ignoreExceptions()
+                .await()
+                .atMost(180, TimeUnit.SECONDS)
+                .untilAsserted(() -> initializeJdbcConnection());
+        initializeJdbcTable();
+        batchInsertData();
+    }
+
+    @Test
+    public void testJdbcGreenplumSourceAndSink() throws IOException, InterruptedException, SQLException {
+        Container.ExecResult execResult = executeSeaTunnelFlinkJob("/jdbc/jdbc_greenplum_source_and_sink.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+
+        // query result
+        String sql = "select age, name from sink";
+        List<Object> result = new ArrayList<>();
+        try (Statement statement = jdbcConnection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                result.add(Arrays.asList(
+                        resultSet.getInt("age"),
+                        resultSet.getString("name")));
+            }
+        }
+        Assertions.assertEquals(100, result.size());
```

Comment:
Only check the amount of data ? Is it possible to add data type and value validation